File: ArtworkPostServices/src/main/java/com/artauction/artworkpostservices/Controllers/TagController.java
package com.artauction.artworkpostservices.Controllers;

import com.artauction.artworkpostservices.Dtos.Tag.TagDto;
import com.artauction.artworkpostservices.Services.Interfaces.TagServices;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/Tag")
@RequiredArgsConstructor
public class TagController {
    // Attributes
    private final TagServices tagServices;

    // Get all tags API
    @GetMapping("/GetAllTags")
    public ResponseEntity<?> getAllTags() {
        return ResponseEntity.ok(tagServices.getAllTags());
    }
}

File: Project/Backend/src/ArtAuction.API/Controllers/TagController.cs
using ArtAuction.Application.Interfaces.Services;
using Microsoft.AspNetCore.Mvc;

namespace ArtAuction.API.Controllers;

[ApiController]
[Route("[controller]")]
public class TagController : ControllerBase
{
    // Attributes
    private readonly ITagServices _tagServices;
    
    // Constructor
    public TagController(ITagServices tagServices)
    {
        this._tagServices = tagServices;
    }

    // Get all tags API
    [HttpGet("GetAllTags")]
    public async Task<IActionResult> GetAllTags()
    {
        // Return all tags
        return Ok(await _tagServices.GetAllTags());
    }
}
